packages/http/src/Session/VerifyCsrfMiddleware.php

```diff
@@ -67,9 +69,14 @@ private function ensureTokenMatches(Request $request): void
     {
         $tokenFromRequest = $request->get(
             key: Session::CSRF_TOKEN_KEY,
-            default: $request->headers->get(self::CSRF_HEADER_KEY),
         );
 
+        if (! $tokenFromRequest && $request->headers->has(self::CSRF_HEADER_KEY)) {
+            $tokenFromRequest = $this->encrypter->decrypt(
```

Comment:
We'd need to `try/catch` this because `decrypt` can throw, in which case the header should be ignored instead of the request failing